subiquitycore/signals.py

```diff
@@ -54,8 +54,8 @@ def connect_signals(self, signal_callback):
             raise SignalException(
                 "Passed something other than a required list.")
         for sig, cb in signal_callback:
-            if sig not in self.known_signals:
-                self.register_signals(sig)
+            # if sig not in self.known_signals:
```

Comment:
It feels silly to only nitpick this and not find much else; but why comment out code? Git history remains.

Reply:
It's good to call this out because I didn't understand what was going on here at all. Turns out urwid.register_signals(cls, signals) *replaces* the registered set of signals for cls, rather than appending them which the original author of this code clearly (reasonably!) expected. You've prompted me to figure this out so I've fixed it properly now, thanks :)